src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts

```diff
@@ -177,21 +180,19 @@ export class TelemetryComponent extends CdForm implements OnInit {
     return result;
   }
 
-  private updateChannelsInReport(updatedConfig: Object = {}) {
-    const channels: string[] = this.report['report']['channels'];
+  private updateReportFromConfig(updatedConfig: Object = {}) {
     const availableChannels: string[] = this.report['report']['channels_available'];
     const updatedChannels = [];
     for (const channel of availableChannels) {
       const key = `channel_${channel}`;
-      // channel unchanged or toggled on
-      if (
-        (!updatedConfig.hasOwnProperty(key) && channels.includes(channel)) ||
-        updatedConfig[key]
-      ) {
+      if (updatedConfig[key]) {
         updatedChannels.push(channel);
       }
     }
     this.report['report']['channels'] = updatedChannels;
+    for (const contactInfoFeild of this.contactInfoFeilds) {
+      this.report['report'][contactInfoFeild] = updatedConfig[contactInfoFeild];
+    }
   }
```

Comment:
Ident checkbox part is working well-- I tried refreshing the page when the ident box was checked, and the contact info box remained aligned with the checkbox! I also see the new "Organization" field, which looks good.

Only thing I'm noticing now is that the `channels` field is not getting updated correctly, which I think is getting handled in this function. If the basic, crash, and device channels are checked, the report preview should list these channels in the `channels` field:
```
{
  "report": {
    "leaderboard": false,
    "report_version": 1,
    "report_timestamp": "2022-02-15T21:46:00.292273",
    "report_id": "68b48dcc-6592-46a9-9b42-9d0ba4579556",
    "channels": [
      "basic",
      "crash",
      "device"
    ],
    "channels_available": [
      "basic",
      "ident",
      "crash",
      "device",
      "perf"
    ],
```
However, it's showing up as an empty list for me now:
```
{
  "report": {
    "leaderboard": false,
    "report_version": 1,
    "report_timestamp": "2022-02-15T21:46:00.292273",
    "report_id": "68b48dcc-6592-46a9-9b42-9d0ba4579556",
    "channels": [],
    "channels_available": [
      "basic",
      "ident",
      "crash",
      "device",
      "perf"
    ],
```

Reply:
@ljflores Thanks for reporting this bug. This was because of the `this.configForm.pristine` check. Anyways its fixed now. :)